### PR TITLE
 Use resource files for messages from the plugins.

### PR DESCRIPTION
**Pull request by dvirZion**
Realization in trivia

### CODE REVIEW

File: Plugins/Trivia/TriviaPlugin.cs [hunk 1]
```diff
@@ -1,9 +1,13 @@
 ﻿using System;
+using System.Reflection;
 using System.Text.Json;
 using BasePlugin.Interfaces;
 using BasePlugin.Records;
 
 
+
```
Comment (YacovGold) [inline Plugins/Trivia/TriviaPlugin.cs]: Extra enters

File: Plugins/Trivia/TriviaPlugin.cs [hunk 2]
```diff
@@ -71,23 +74,23 @@ public void Execute(PluginInput input)
                     else if (input.Message.Length > 1 || input.Message[0] < '1' || input.Message[0] > '4')
                     {
                         input.Callbacks.SavePluginUserData(input.PersistentData);
-                        input.Callbacks.SendMessage("Your answer is invalid, You need to enter a number between 1 and 4");
+                        input.Callbacks.SendMessage(Resources.Plugins.Trivia_Eror2);
```
Comment (YacovGold) [inline Plugins/Trivia/TriviaPlugin.cs]: Error2? please give a better name

File: Plugins/Trivia/TriviaPlugin.cs [hunk 3]
```diff
@@ -96,29 +99,29 @@ public void Execute(PluginInput input)
...
-            nums = rndNums(2, 10, true);
+            nums = RndNums(2, 10, true);
             result = nums[0] * nums[1];
-            allResult = rndNums(4, 100, false);
+            allResult = RndNums(4, 100, false);
             Random rnd = new Random();
             trueResult = rnd.Next(4);
             allResult.SetValue(result, trueResult);
-            var data = ($"what the result of: {nums[0]} * {nums[1]} =\n");
+            var data = ($"{Resources.Plugins.Trivia_Question} {nums[0]} * {nums[1]} =\n");
```
Comment (YacovGold) [inline Plugins/Trivia/TriviaPlugin.cs]: The text in the resource should have the {}
For example
"what the result of: {0} * {1} =\n"
then do something like this
`string.format(Resources.Plugins.Trivia_Question, nums[0], nums[1])`

File: Plugins/Trivia/TriviaPlugin.cs [hunk 4]
```diff
@@ -96,29 +99,29 @@ public void Execute(PluginInput input)
...
             for (int i = 0; i < allResult.Length; i++)
             {
                 data += ($"{i + 1}) {allResult[i]}   ");
             }
             return data;
         }
 
-        private int[] rndNums(int numResult, int maxOfNum, bool repeat)
+        private int[] RndNums(int numResult, int maxOfNum, bool repeat)
```
Comment (YacovGold) [inline Plugins/Trivia/TriviaPlugin.cs:119]: Nice catch

File: Runners/ConsoleApp/.filenesting.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "help":"https://go.microsoft.com/fwlink/?linkid=866610"
```
Comment (YacovGold) [inline Runners/ConsoleApp/.filenesting.json]: What is this file?

File: Runners/ConsoleApp/ConsoleApp.csproj [hunk 6]
```diff
@@ -5,6 +5,15 @@
     <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
+    <COMReference Include="{215d64d2-031c-33c7-96e3-61794cd1ee61}">
+      <WrapperTool>tlbimp</WrapperTool>
```
Comment (YacovGold) [inline Runners/ConsoleApp/ConsoleApp.csproj]: Why reference tlbimp?

**Comment by YacovGold** [comment]
Its not for all files, please change the title

File: Runners/ConsoleApp/.filenesting.json (hunk 5 above)
Comment (dvirZion) [reply Runners/ConsoleApp/.filenesting.json]: Unnecessary.
Probably a result of the attempts to work with the resources

File: Runners/ConsoleApp/ConsoleApp.csproj (hunk 6 above)
Comment (dvirZion) [reply Runners/ConsoleApp/ConsoleApp.csproj]: Unnecessary.
Probably a result of the attempts to work with the resources

File: Plugins/Trivia/TriviaPlugin.cs (hunk 1 above)
Comment (dvirZion) [reply Plugins/Trivia/TriviaPlugin.cs]: ok

File: Plugins/Trivia/TriviaPlugin.cs (hunk 2 above)
Comment (dvirZion) [reply Plugins/Trivia/TriviaPlugin.cs]: ok

File: Plugins/Trivia/TriviaPlugin.cs (hunk 3 above)
Comment (dvirZion) [reply Plugins/Trivia/TriviaPlugin.cs]: This is how it should be written - string.Format(Resources.Plugins.Trivia_Question, nums[0], nums[1],"\n");
i performed.

File: Plugins/Trivia/TriviaPlugin.cs (hunk 4 above)
Comment (dvirZion) [reply Plugins/Trivia/TriviaPlugin.cs:119]: Thanks

**Comment by dvirZion** [comment]
Correcting the problems as I was asked

**Comment by dvirZion** [comment]
now ok?

**Pull request merged**